src/features/island/buildings/components/building/bakery/Bakery.tsx

```diff
@@ -33,8 +35,11 @@ export const Bakery: React.FC<Props> = ({
   const [showModal, setShowModal] = useState(false);
   const { setToast } = useContext(ToastContext);
 
-  if (!craftingService || !handleShowCraftingTimer)
+  useUiRefresher({ active: crafting });
```

Comment:
Why do we need this ui refresher here? We have actually built these components to NOT refresh on every tick but to instead only rerender when a state changes. We have done this by interpreting the machine and only subscribing to certain changes. As the machine has a tick implemented we could easily just read the machine and get the rerender out of the box.

Why is there a need for it?

Reply:
The `useUiRefresher` was implemented because the progress bar won't update without the refresher.

I have updated the PR to consume the `secondsTillReady` properly for the buildings, so I have removed the `useUiRefresher` refrences from these buildings (see my latest commit)